tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -328,26 +328,33 @@ Internal::FriendInfo TTreeProcessorMT::GetFriendInfo(TTree &tree)
 
    for (auto fr : *friends) {
       const auto frTree = static_cast<TFriendElement *>(fr)->GetTree();
+      const bool isChain = frTree->IsA() == TChain::Class();
 
-      // Check if friend tree has an alias
-      const auto realName = frTree->GetName();
-      const auto alias = tree.GetFriendAlias(frTree);
-      if (alias) {
-         friendNames.emplace_back(std::make_pair(realName, std::string(alias)));
-      } else {
-         friendNames.emplace_back(std::make_pair(realName, ""));
-      }
```

Comment:
Couldn't moving this to a branch where `isChain` is false break alias support (whatever we currently support) for TChain friends?

Reply:
You spotted the issue of the failing tests above :) I've move it out of the branching and we support (again) aliases for chains.

Reply:
I think the new logic is wrong, `GetFriendAlias` returns a `const char *`, potentially a `nullptr`, and if I read correctly that is passed to a `std::string(alias)` with no check at the moment...?

Can we go back to the original logic but saving `alias || ""` in a local variable instead of appending to `friendNames` directly?

Reply:
Sry you are right. I haven't looked up what the `GetFriendAlias` returns :/

Fixed!

Reply:
I think it's still wrong below, line 358?

Can we go back to the original logic but saving `alias || ""` in a local variable instead of appending to `friendNames` directly?

Reply:
Today is not my day :see_no_evil: I've fixed also the logic there now :/

Reply:
Man I'm sorry but now the logic is repeated twice and it's far from where `alias` is assigned -- other people in the future could commit this same mistake. Can you reply to my question above, how about going back to the original `alias` handling logic but saving alias-or-"" in a local variable instead of appending to friendNames directly?

For example:

```c++
const char *alias_c = tree.GetFriendAlias(frTree);
const std::string alias = alias_c != nullptr ? alias_c : "";
```

and below just

```c++
friendNames.emplace_back(std::make_pair(realName, std::string(alias)))
```

So it's clear to the reader that `alias_c` can be a nullptr, and that case is handled right there and then instead of replicating the logic twice below.

Reply:
Sry haven't read carefully your comment :grimacing: But I appreciate that you review carefully :) Now!

Reply:
hooray! :smile: thanks for bearing with me